Wazo 3286 api

franckrupin opened the pull request.

franckrupin committed:
Add srvice definition

franckrupin committed:
Add Makefile

franckrupin committed:
Add plugins and configurations

franckrupin committed:
Add tests folder

franckrupin committed:
Fix ssl in pilot

franckrupin committed:
Fix cli

franckrupin committed:
Fix api

franckrupin committed:
Fix pilot

franckrupin committed:
Rename folder

franckrupin committed:
Fix tox

pc-m commented on wazo_load_gateway/http_server.py:
```diff
@@ -0,0 +1,38 @@
+        # and if we have more than one
+        # they will each get their own queue and then not respect the execution order
+        # which creates concurrency issues.
+        self.cfg.set('workers', 1)
```
Is it really a problem to have multiple worker process in this service?

franckrupin replied:
I can't really say for this part, I rely on previous work and if a concurrency has been detected it could be difficult to reproduce, but I'll have a look.

pc-m commented:
recheck

pc-m commented:
recheck

pc-m commented on wazo_load_api/config.py:
```diff
@@ -0,0 +1,69 @@
+        'private_key': '/etc/wazo-load-api/private.key',
+        'cors': {
+            'enabled': True,
+            'allow_headers': ['Content-Type', 'X-Auth-Token'],
```
`X-Auth-Token` is not required here since we are not using token authentication

pc-m commented on wazo_load_api/etc/wazo-load-api/private.key:
```diff
@@ -0,0 +1,28 @@
+-----BEGIN PRIVATE KEY-----
```
Is it a concern that this file is public?

franckrupin replied:
Not a problem due to the fact we generate it on demand for testing purpose, this one particularly won't be used anymore.

pc-m commented on wazo_load_api/plugins/job/http.py:
```diff
@@ -0,0 +1,22 @@
+    try:
+        command = payload["cmd"]
+    except KeyError as e:
+        return {"status": "internal_error", "error": str(e)}
```
I think the proper way to return an error with fastapi is to raise an HTTPException
```python
from fastapi import HTTPException
raise HTTPException(status_code=400, detail={'status': 'internal_error', 'error': str(e)})
```

pc-m commented on wazo_load_api/plugins/job/services.py:
```diff
@@ -0,0 +1,18 @@
+
+
+async def run(cmd):
+    print(cmd)
```
should be a logger.debug

franckrupin replied:
yeah I missed this one.

pc-m commented on wazo_load_api/plugins/job/tests/test_http.py:
```diff
@@ -0,0 +1,55 @@
+# Créer un client de test pour le routeur
+client = TestClient(router)
+
+def test_run_load_success():
```
Can you add unit tests to `zuul.yaml`

pc-m commented on wazo_load_api/plugins/job/tests/test_http.py:
```diff
@@ -0,0 +1,55 @@
+#    response = client.post("/run", json=command)
+
+#    # Assurer que la réponse a un statut HTTP 200 et que le contenu est conforme à une erreur interne
+#    assert response.status_code == 200
```
This test should be uncommented and the status_code should be a `400` for a bad payload

pc-m commented on wazo_load_api/plugins/job/tests/test_http.py:
```diff
@@ -0,0 +1,55 @@
+#    response = client.post("/run", json=command)
+
+#    # Assurer que la réponse a un statut HTTP 200 et que le contenu est conforme à une erreur interne
+#    assert response.status_code == 200
```
I think this one should be a 500

pc-m commented on wazo_load_api/plugins/job/http.py:
```diff
@@ -0,0 +1,22 @@
+router = APIRouter()
+
+
+@router.post("/run")
```
`api.yml` is missing for the job plugin

pc-m commented on wazo_load_api/setup.py:
```diff
@@ -0,0 +1,28 @@
+        'console_scripts': [
+            f'{NAME}=main:main',
+        ],
+        'plugins': [
```
The entry point should be something more unique since any python executable could use the entry point.
I suggest `wlapi-plugins`

franckrupin committed:
Remove useless comments

franckrupin committed:
Remove certs artifacts

franckrupin committed:
Address PR's comments

franckrupin committed:
Improve pilot

- Add environment propagtion for application setup
- Fix tests for new code
- Add configuration optimization for reaching a load cluster
- Modify docker build and makefile

franckrupin committed:
Modify load generator

- Remove useless code due to the clusterizaion of the load nodes
- Add sections in configuration file
- Modify the load configuration file example

franckrupin committed:
Modify api

- Refactor the module
- Fix tests
- Add environement variables to be propagated to the command

franckrupin committed:
Fix linters and tests

franckrupin committed:
Fix typo

pc-m committed:
plugins: rename namespaces

avoid using the same namespace for both projects

pc-m committed:
add a gitignore
